src/rendering/utils/GraphicUtils.ts

```diff
@@ -6,6 +6,30 @@ import datamath from '@lib/datamath'
 import { Graphics, LineStyle, Text, TextStyle } from '@lib/pixi'
 import { Sprite, Texture, Matrix, Container, gsap } from '@lib/pixi'
 
+export class CoveredText extends Graphics {
+
+    public update(updater: (textGraphic: Text) => void, position: [number, number], style: { padding, anchor }): void {
```

Comment:
very compicated interface of the update function

funciton,
array,
object

and In general I don't really like object oriented approach inside renderer and I think we should avoid it if possible and create utility funcitons instead.